### ISSUE TITLE
Tableau is bugged in Debugging Mode on Mac OSX

### ISSUE THREAD

**Issue by merlijnbuit** [tableau/extensions-api#287]
Hi,

I am working with Tableau on Mac OS now (Yay!)
But opening Tableau Desktop in debug mode:

open /Applications/Tableau\ Desktop\ 2019.4.app --args --remote-debugging-port=8696

results in a buggy Tableau. Existing workbooks can't be opened and give errors.
Opening a empty workbook and connecting to a datasource will result in white labels in many places. 

I am using Tableau 2019.4 I think this issue is already present since 2018.2.

Any fixes? :)

Merlijn
<img width="538" alt="Schermafbeelding 2019-12-02 om 20 29 12" src="https://user-images.githubusercontent.com/16118919/69988707-7600a500-1542-11ea-918d-6d12f5ecf269.png">
<img width="433" alt="Schermafbeelding 2019-12-02 om 20 29 28" src="https://user-images.githubusercontent.com/16118919/69988708-7600a500-1542-11ea-9044-9ff69fd53874.png">

**Comment by johnDance** [comment]
Hi Merlijn:

We cannot duplicate this locally. Can you give us some more information?
1. What version of MacOS are you using?
2. What happens if you just run: open /Applications/Tableau\ Desktop\ 2019.4.app?
(I'm grasping at straws here...)
3. Is there better behavior if you run: /Applications/Tableau\ Desktop\ 2019.4.app/Contents/MacOS/Tableau --remote-debugging-port=8696

Thanks for the additional info.
John

**Comment by merlijnbuit** [comment]
1. Latest version on a brand new Macbook Pro 16". But I get similar results on my iMac @ home.
2. Running in normal mode is fine.
3. Running from 3 gives me these errors in my console:
![Schermafbeelding 2019-12-09 om 09 38 49](https://user-images.githubusercontent.com/16118919/70420283-b5118780-1a67-11ea-853d-ab6d15aa1037.png)

**Comment by johnDance** [comment]
Hi Merlijn:
I have tried to duplicate the issue on Sierra, Mojave, and Catalina, but could not duplicate the buggy Tableau behavior. 

As an aside, while the following gives warnings and a few errors, it should still load and run, and allow you to debug:  /Applications/Tableau\ Desktop\ 2019.4.app/Contents/MacOS/Tableau --remote-debugging-port=8696 

But perhaps the following will work for you. In a terminal, set up an environment variable before running Tableau. This will work as long as you run Tableau from this terminal.

export QTWEBENGINE_REMOTE_DEBUGGING=8696
open /Applications/Tableau\ Desktop\ 2019.4.app

Let us know how this goes.
Thanks,
John

**Comment by merlijnbuit** [comment]
Hi John,

The bug is still occurring, it is pretty hard debugging existing workbooks like this :) 
I managed to retrieve some debug info. Maybe this helps?

![Schermafbeelding 2020-09-15 om 13 40 14](https://user-images.githubusercontent.com/16118919/93206114-3285c900-f759-11ea-85f7-d0ba92287575.png)

**Comment by merlijnbuit** [comment]
My Mac OSX is running in Dutch language.

**Comment by johnDance** [comment]
Wow, that is very strange.  You won't get those errors if you "open <Tableau Path>.app --args --remote-debugging-port=8696", but you already reported that didn't work either. I have one more idea...
Set up a system wide environment variable (see https://stackoverflow.com/questions/135688/setting-environment-variables-on-os-x)
QTWEBENGINE_REMOTE_DEBUGGING=8696

Then just run Tableau normally.
Let me know if that works while running in Dutch.
Thanks,
John

**Comment by lavanyasreepada** [comment]
Hi,

I'm addicted to my MacBook Pro and Tableau ( [Tableau online Training](https://mindmajix.com/tableau-training) ), so I've got the package running nicely on an XP virtual machine (in Fusion). Works fine for me, although I had to goose up the VM memory beyond the 500 MB default setting to get the performance I was used to on my PC desktop.  When those market conditions and engineering resources align I'd love to see Tableau running directly on OS X too.

Thanks and regards,
Lavanya Sreepada
